man/eqr.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH eqr 1  "eqr 1.9.90" 
.SH NAME
eqr \- Encode text into svg/png/jpg/terminal\-format QR codes with optional logo
.SH SYNOPSIS
\fBqr\fR [\fB\-o\fR|\fB\-\-output\fR] [\fB\-F\fR|\fB\-\-force\fR] [\fB\-t\fR|\fB\-\-terminal\fR] [\fB\-l\fR|\fB\-\-level\fR] [\fB\-p\fR|\fB\-\-path\fR] [\fB\-P\fR|\fB\-\-proportion\fR] [\fB\-e\fR|\fB\-\-edge\fR] [\fB\-f\fR|\fB\-\-fg\fR] [\fB\-b\fR|\fB\-\-bg\fR] [\fB\-s\fR|\fB\-\-scale\fR] [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fISTRING\fR] 
.SH DESCRIPTION
Encode text into svg/png/jpg/terminal\-format QR codes with optional logo
.SH OPTIONS
.TP
\fB\-o\fR, \fB\-\-output\fR \fI<QR_PATH>\fR
Output file (jpg/png/svg) [default: qr.png]
.TP
\fB\-F\fR, \fB\-\-force\fR
Force the overwrite of an existing output file
.TP
\fB\-t\fR, \fB\-\-terminal\fR
Output to terminal (never the logo)
.TP
\fB\-l\fR, \fB\-\-level\fR \fI<LEVEL>\fR [default: M]
QR error correction level (L: 7%, M: 15%, Q: 25%, H: 30%)
.TP
\fB\-p\fR, \fB\-\-path\fR \fI<LOGO_PATH>\fR
Path to logo (png/jpg)
.TP
\fB\-P\fR, \fB\-\-proportion\fR \fI<PROPORTION>\fR [default: 0.25]
Logo proportion to the whole image (0..1)
.TP
\fB\-e\fR, \fB\-\-edge\fR \fI<EDGE>\fR [default: 4]
Edge size (in unit blocks)
.TP
\fB\-f\fR, \fB\-\-fg\fR \fI<FG>\fR [default: 000]
Foreground RGB color (hex code)
.TP
\fB\-b\fR, \fB\-\-bg\fR \fI<BG>\fR [default: fff]
Background RGB color (hex code)
.TP
\fB\-s\fR, \fB\-\-scale\fR \fI<SCALE>\fR [default: 8]
Size of unit block in pixels (1..255)
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version
.TP
[\fISTRING\fR]
String to encode (can also be piped in)
.SH VERSION
v1.9.90


CHANGELOG.md
# Changelog
## [1.7.3] - 2025-03-09
- Fmt and update deps

## [1.7.2] - 2025-02-22
- Update cargo-binstall link

## [1.7.1] - 2025-02-22
- Update CI and tempfile version

## [1.7.0] - 2025-02-22
- Refactor and add promptpay

## [1.6.2] - 2025-02-08
- Fixed readme, modified help

## [1.6.1] - 2025-02-08
- Fixed bad path error

## [1.6.0] - 2025-02-04
- Proportion option for logo

## [1.5.2] - 2025-02-04
- Allow logo transparency

## [1.5.1] - 2025-01-30
- Double logo size

## [1.5.0] - 2025-01-30
- Allow logo overlay

## [1.4.3] - 2025-01-13
- Set single dependency version

## [1.4.2] - 2025-01-13
- Fix dependency version

## [1.4.1] - 2025-01-13
- Update dependencies
- Changelog

## [1.4.0] - 2025-01-12
- Update dependencies
- Compact help 

## [1.3.0] - 2025-01-12
- Default output file name
- Terminal option

## [1.2.0] - 2025-01-11
- Redo options

## [1.1.5] - 2024-11-13
- Update deps

## [1.1.4] - 2024-10-29
- Repo

## [1.1.3] - 2024-10-29
- Cargo-binstall

## [1.1.2] - 2024-10-23
- Update deps

## [1.1.1] - 2024-10-19
- Enhance doc

## [1.1.0] - 2024-10-08
- Completions

## [1.0.1] - 2024-10-05
- Correct manpage

## [1.0.0] - 2024-10-05
### Changed
- Repo name to eqr
- Pushed to crates.io

## [0.3.2] - 2024-09-05
### Changed
- Static (musl) binary

## [0.3.1] - 2024-09-02
### Added
- Release binary

## [0.3.0] - 2024-09-02
### Added
- Short flag `-L` for `--error-correction-level`
- Atrributes width/height on svg element for scale

### Changed
- Repo at github.com/pepa65/qr
- Allow scale for svg
- Fix border on raster images
- Fix help on fg/bg
- No inverted colors on terminal
- No `#` to specify hex colors

### Removed
- xml/DOCTYPE headers from svg
- Tabs from svg
- Attribute stroke from svg
- `jpeg` file format from `-o`/`--output`

## [0.2.0] - 2022-07-18
### Added
- Feature to read string to encode from standard input, which allows to pipe
  commands to `qr-rs` (closing issue #1).
- `border` CLI option to specify border size.
- `error-correction-level` CLI option to specify QR *error-correction-level* as
  one of the following values:
  - **low**;
  - **medium**;
  - **quartile**;
  - **high**.

## [0.1.0] - 2022-07-15
Initial release.


completions/eqr.bash
_eqr() {
    local i cur prev opts cmd
    COMPREPLY=()
    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
        cur="$2"
    else
        cur="${COMP_WORDS[COMP_CWORD]}"
    fi
    prev="$3"
    cmd=""
    opts=""

    for i in "${COMP_WORDS[@]:0:COMP_CWORD}"
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="eqr"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        eqr)
            opts="-o -F -t -l -p -P -e -f -b -s -h -V --output --force --terminal --level --path --proportion --edge --fg --bg --scale --help --version"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --output)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --level)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -l)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --path)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --proportion)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -P)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --edge)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -e)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --fg)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -f)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --bg)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --scale)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _eqr -o nosort -o bashdefault -o default eqr
else
    complete -F _eqr -o bashdefault -o default eqr
fi
